src/Adapter/Product/Combination/Update/CombinationStockUpdater.php

```diff
@@ -138,54 +138,45 @@ private function fillUpdatableProperties(Combination $combination, CombinationSt
      */
     private function updateStockAvailable(Combination $combination, CombinationStockProperties $properties): void
     {
-        $updateQuantity = null !== $properties->getQuantity();
         $updateLocation = null !== $properties->getLocation();
+        // We would skip zero delta quantity, so we check if its truthy instead of checking strictly against null.
+        $deltaQuantity = $properties->getDeltaQuantity();
 
-        if (!$updateQuantity && !$updateLocation) {
+        if (!$deltaQuantity && !$updateLocation) {
             return;
         }
 
-        $newQuantity = $properties->getQuantity();
-        $newLocation = $properties->getLocation();
-
         $stockAvailable = $this->stockAvailableRepository->getForCombination(new CombinationId((int) $combination->id));
 
-        if ($updateQuantity) {
-            $this->saveMovement($combination, (int) $stockAvailable->quantity, $newQuantity);
-            $stockAvailable->quantity = $newQuantity;
+        if ($deltaQuantity) {
+            $stockAvailable->quantity += $deltaQuantity->getDeltaQuantity();
         }
 
         if ($updateLocation) {
-            $stockAvailable->location = $newLocation;
+            $stockAvailable->location = $properties->getLocation();
         }
 
         $this->stockAvailableRepository->update($stockAvailable);
+
+        // save movement only after stockAvailable has been updated
+        if ($deltaQuantity) {
+            $this->saveMovement($stockAvailable, $deltaQuantity);
+        }
     }
 
     /**
-     * @param Combination $combination
-     * @param int $oldQuantity
-     * @param int $newQuantity
-     *
-     * @throws CoreException
+     * @param StockAvailable $stockAvailable
+     * @param DeltaQuantity $deltaQuantity
      */
-    private function saveMovement(Combination $combination, int $oldQuantity, int $newQuantity): void
+    private function saveMovement(StockAvailable $stockAvailable, DeltaQuantity $deltaQuantity): void
```

Comment:
We should check that `$deltaQuantity->getDeltaQuantity() !== 0` or we may end up with empty stock movements. There is no validation in the VO that prevents a 0 input, which is not a problem I guess but then we need to check the value before applying it

Reply:
Or the alternative can be to prevent 0 as a DeltaQuantity in the VO, which is not such a bad idea an 0 delta is like no delta at all So the builder will need to check if the value is different from zero (the check is performed sooner) But at least when a `DeltaQuantity` VO is present we are sure it contains a modification

Reply:
probably the constraint in VO is better